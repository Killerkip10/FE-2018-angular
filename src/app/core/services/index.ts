export * from './auth.service';
export * from './profile.service';
export * from './token.service';
export * from './search-user.service';
