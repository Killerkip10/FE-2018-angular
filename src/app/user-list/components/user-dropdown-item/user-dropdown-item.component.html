<mat-card>
  <mat-card-title>{{user?.name}}</mat-card-title>
  <mat-card-subtitle>{{user?.login}}</mat-card-subtitle>

  <mat-card-content>
    <div>{{user?.age}}</div>
    <div>{{user?.dateOfBirth | date: 'medium'}}</div>
    <div>{{user?.dateOfFirstLogin | date: 'medium'}}</div>
    <div>{{user?.dateOfNextNot | date: 'medium'}}</div>
    <div>{{user?.information}}</div>
  </mat-card-content>
</mat-card>
