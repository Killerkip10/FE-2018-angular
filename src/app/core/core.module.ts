import {NgModule} from '@angular/core';
import {BrowserModule} from '@angular/platform-browser';
import {TranslateModule} from '@ngx-translate/core';
import {HttpClientModule} from '@angular/common/http';
import {RouterModule} from '@angular/router';

import {BrowserAnimationsModule} from '@angular/platform-browser/animations';
import {AuthGuard} from './guards';
import {AuthService} from './services';
import {
  MatButtonModule,
  MatToolbarModule,
  MatSelectModule,
  MatIconModule
} from '@angular/material';
import {
  HeaderComponent,
  FooterComponent
} from './components';

@NgModule({
  imports: [
    BrowserModule,
    HttpClientModule,
    TranslateModule,
    BrowserAnimationsModule,

    RouterModule,

    MatIconModule,
    MatButtonModule,
    MatSelectModule,
    MatToolbarModule
  ],
  declarations: [
    HeaderComponent,
    FooterComponent
  ],
  providers: [
    AuthService,

    AuthGuard
  ],
  exports: [
    HeaderComponent,
    FooterComponent
  ]
})
export class CoreModule {}
