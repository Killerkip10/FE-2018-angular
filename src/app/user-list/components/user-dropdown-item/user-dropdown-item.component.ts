import {Component, Input} from '@angular/core';

import {ShortUser} from '../../../models';

@Component({
  selector: 'app-user-dropdown-item',
  templateUrl: './user-dropdown-item.component.html',
  styleUrls: ['./user-dropdown-item.component.scss']
})
export class UserDropdownItemComponent {
  @Input() public user: ShortUser;

  constructor() { }
}
