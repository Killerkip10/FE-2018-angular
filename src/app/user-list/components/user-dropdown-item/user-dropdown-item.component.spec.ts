import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import {NO_ERRORS_SCHEMA} from '@angular/core';

import { UserDropdownItemComponent } from './user-dropdown-item.component';

describe('ItemComponent', () => {
  let component: UserDropdownItemComponent;
  let fixture: ComponentFixture<UserDropdownItemComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [
        UserDropdownItemComponent
      ],
      schemas: [
        NO_ERRORS_SCHEMA
      ]
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(UserDropdownItemComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });
});
