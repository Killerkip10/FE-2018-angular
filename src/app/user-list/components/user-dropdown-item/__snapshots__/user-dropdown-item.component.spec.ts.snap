// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDropDownItemComponent snapshot 1`] = `
<app-user-dropdown-item>
  <mat-card>
    <mat-card-title>
      
    </mat-card-title>
    <mat-card-subtitle>
      
    </mat-card-subtitle>
    <mat-card-content>
      <div>
        
      </div>
      <div>
        
      </div>
      <div>
        
      </div>
      <div>
        
      </div>
      <div>
        
      </div>
    </mat-card-content>
  </mat-card>
</app-user-dropdown-item>
`;

exports[`UserDropDownItemComponent snapshot 2`] = `
<app-user-dropdown-item
  user={[Function Object]}
>
  <mat-card>
    <mat-card-title>
      
    </mat-card-title>
    <mat-card-subtitle>
      
    </mat-card-subtitle>
    <mat-card-content>
      <div>
        
      </div>
      <div>
        
      </div>
      <div>
        
      </div>
      <div>
        
      </div>
      <div>
        
      </div>
    </mat-card-content>
  </mat-card>
</app-user-dropdown-item>
`;

exports[`UserDropDownItemComponent snapshot 3`] = `
<app-user-dropdown-item
  user={[Function Object]}
>
  <mat-card>
    <mat-card-title>
      Pasha
    </mat-card-title>
    <mat-card-subtitle>
      pasha
    </mat-card-subtitle>
    <mat-card-content>
      <div>
        25
      </div>
      <div>
        Nov 10, 2018, 9:52:23 PM
      </div>
      <div>
        Nov 10, 2018, 9:52:23 PM
      </div>
      <div>
        Nov 10, 2018, 9:52:23 PM
      </div>
      <div>
        pahan
      </div>
    </mat-card-content>
  </mat-card>
</app-user-dropdown-item>
`;
